### DISCUSSION TITLE
更新日志 - 22.03.5 - 20230908

### DISCUSSION THREAD

**Discussion by jjm2473** [istoreos/istoreos#994] [Announcements]
**同时带有光口和电口的x86机型用户注意备份配置，升级可能导致网口顺序变化**

# 系统
1. 增加 binfmt-misc 文件系统，以支持 Docker 跨平台运行镜像，例如在 x86 机型上运行 arm 的镜像。使用方法参考后面的链接
2. 修复从 21 版本固件升级以后 Docker 自带防火墙未自动切换到系统防火墙的问题

# 设备
* x86: 
  1. 单网口机型 LAN 口自动 NAT，改为只针对 Docker 默认网络
  2. 修改部分网卡驱动的加载优先级，尽量使电口先于光口加载，也就是尽量使 eth0 是电口。 **同时带有光口和电口的机型用户注意，升级可能导致网口顺序变化，导致wan口与某个lan口交换，可能需要手动修改 /etc/config/network 修复**

# 参考链接
1. Docker 跨平台运行镜像 https://docs.docker.com/build/building/multi-platform/#qemu

**Comment by kirafreedom** [comment]
升级后docker容器都无法联网怎么破？

**Reply by jjm2473** [reply]
单网口设备？在防火墙勾选LAN区域的IP伪装，或者恢复出厂设置

**Comment by yuanxiang2015** [comment]
你好，我的物理机是J4125的，然后我有一张BCM94352HBM的无线网卡，请问怎么才能开启它的WiFi功能呢？
lspci 看到的硬件信息是
06:00.0 Network controller: Broadcom Inc. and subsidiaries BCM4352 802.11ac Wireless Network Adapter (rev 03)

**Reply by jjm2473** [reply]
如果知道驱动名称，软件包那边可以搜索安装驱动。如果不知道驱动名称，试试openwrt官方固件能不能驱动，如果不行，那就可能openwrt没这个驱动。

**Comment by TommyLiu-1985** [comment]
这个版本升级上来后，内存占用大幅增加，我原本分配1GB内存都绰绰有余的，现在得分配2GB才勉强够了。不知道是不是个例。

**Reply by jjm2473** [reply]
有没有使用docker？执行`htop`，然后点一下`RES`列，可以看内存占用排序

**Reply by TommyLiu-1985** [reply]
没有使用docker

![image](https://github.com/istoreos/istoreos/assets/82349643/aff67a58-8217-4725-8087-4c4c03ec1e52)

**Reply by jjm2473** [reply]
看起来内存占用很少啊，等你觉得内存不对劲了再执行一次

**Reply by xpn282** [reply]
一摸一样的现象，升级后，内存占用大幅上升！
而且首页老是，提示软件源配置错误。。。
没升级之前这些 都是正常的。。。

**Reply by TommyLiu-1985** [reply]
是的 ，我也提示软件源配置错误，从清华的源换了其他的就可以。

**Reply by xpn282** [reply]
作者修复一下这些问题吧，不然看着都不爽。。。。内存问题，软件源问题。。。。

**Reply by TommyLiu-1985** [reply]
![image](https://github.com/istoreos/istoreos/assets/82349643/1722370b-adfa-4c9e-8199-8bdb5b3d8ff6)
刷回之前一个版本的内存占用对比，相差大概500M！

**Reply by jjm2473** [reply]
@TommyLiu-1985 ~~哈？你这个不是只占用了214M吗，哪来的差距500M？~~ 哦，截图是之前版本

**Comment by LinkAiR** [comment]
![Screenshot_2023-09-14-20-13-18-045_com microsoft emmx-edit](https://github.com/istoreos/istoreos/assets/39733469/142620d5-99dd-46a6-b689-9839aa850f2c)
升级后内存巨幅上升，1G内存完全不够用。

**Reply by xpn282** [reply]
内存占用起码多出了60%这样。。。。固件还是别乱升级比较好。如果没有特别需求，还是别折腾

**Reply by jjm2473** [reply]
执行htop，然后点RES列，看看是什么占用的内存

**Comment by k2y1982** [comment]
我的内存直接翻倍了。上个版本30%，现在60%，没办法只好退回上个版本了。

**Comment by jjm2473** [comment]
x86和rk35xx内存占用增高的原因找到了，是这个版本新增加的 erofs 文件系统模块，之后会移除或修复此模块。

**Reply by hsiangkao** [reply]
什么意思，目前你们使用了erofs的什么功能？

**Reply by jjm2473** [reply]
并没有使用erofs什么功能，只是因为erofs在op官方的仓库还没有（23版本以上才有），用户自己想安装也安装不了，所以集成下。很多文件系统都是集成了也不一定用得上，例如hfs，jfs，你看看 `/proc/filesystem` 有多少文件系统是你知道的。我自己用erofs是为了挂载安卓的镜像文件，做研究。

**Reply by hsiangkao** [reply]
但这个我理解不影响rss这个，所以感觉比较奇怪。
一个文件系统模块，即使insmod加载（其实也可以动态加载），如果没有挂载，基本常驻代码像erofs也就100k。

**Reply by jjm2473** [reply]
@hsiangkao 是不影响用户空间的RSS，而是模块自己就占用了512M内存，用户空间是看不到的。之前我是以为用户空间的程序占用了内存，才说用htop检查。模块自己占用的内存要用lsmod命令才能检查。
5.10内核的erofs有个 `EROFS_FS_CLUSTER_PAGE_LIMIT` 的参数，这个参数越大，占用的内存就越多，而且是只要加载模块就占用（没挂载），这个参数每增加1大概就要占用2MB内存。5.15以后的内核没有这个参数了，erofs模块只会占用几百k的内存。

**Reply by hsiangkao** [reply]
EROFS_FS_CLUSTER_PAGE_LIMIT在5.10是个没用的参数，只能是1，调别的也没用

**Reply by jjm2473** [reply]
@hsiangkao 有，范围是1到256，你调到256试试，就会占用512M内存了

**Reply by jjm2473** [reply]
这是代码文档：
```
config EROFS_FS_CLUSTER_PAGE_LIMIT
	int "EROFS Cluster Pages Hard Limit"
	depends on EROFS_FS_ZIP
	range 1 256
	default "1"
	help
	  Indicates maximum # of pages of a compressed
	  physical cluster.

	  For example, if files in a image were compressed
	  into 8k-unit, hard limit should not be configured
	  less than 2. Otherwise, the image will be refused
	  to mount on this kernel.
```

看文档意思是PAGE_SIZE是4KB，但是实际是2MB，乘上 EROFS_FS_CLUSTER_PAGE_LIMIT 就是很大的内存了。

**Reply by jjm2473** [reply]
https://elixir.bootlin.com/linux/v5.10.176/A/ident/CONFIG_EROFS_FS_CLUSTER_PAGE_LIMIT

**Reply by hsiangkao** [reply]
我说的是功能上没作用，所以只有1有作用

**Comment by Ginua8292** [comment]
R4S太卡了！4G内存版本！用起来一点都不流畅！还开IPV6后，有几率IPV6会死掉！IPV4有IP也打开不了网络！

**Reply by jjm2473** [reply]
你确定不是浏览器不行？

**Reply by Ginua8292** [reply]
不是！我自己编了个！也是istore的界面！或许和官方的就是用的是lede源码和没有某些功能！但一点也不卡！

**Reply by jjm2473** [reply]
你是不是把翻墙插件的节点列表页面作为判断标准了？
